packages/react-components/react-charts-preview/src/utilities/FocusableTooltipText.tsx

```diff
@@ -1,80 +1,66 @@
-  private _async: Async;
-
-  constructor(props: IFocusableTooltipTextProps) {
-    super(props);
-
-    this.state = {
-      textOverflow: false,
-    };
+export const FocusableTooltipText:  React.FunctionComponent<IFocusableTooltipTextProps> = React.forwardRef<
+HTMLDivElement,
+IFocusableTooltipTextProps
+>((props, forwardedRef) => {
+  const [textOverflow, setTextOverflow] = useState(false);
+  const tooltipChild = useRef<HTMLSpanElement>(null);
+  const async = useRef(new Async()).current;
+  const resizeObserver = useRef<ResizeObserver>();
 
-    this._async = new Async(this);
-  }
-
-  public render(): React.ReactNode {
-    const { className, content, accessibilityData } = this.props;
-
-    return (
-      <TooltipHost overflowMode={TooltipOverflowMode.Self} hostClassName={className} content={content}>
-        <span
-          {...getAccessibleDataObject(accessibilityData)}
-          ref={this._tooltipChild}
-          data-is-focusable={this.state.textOverflow}
-        >
-          {content}
-        </span>
-      </TooltipHost>
-    );
-  }
+  const getTargetElement = useCallback((): HTMLElement | undefined => {
+    if (!tooltipChild.current || !tooltipChild.current.parentElement) {
+      return undefined;
+    }
+    return tooltipChild.current.parentElement;
+  }, [tooltipChild]);
 
-  public componentDidMount(): void {
-    const overflowElement = this._getTargetElement();
-    if (window.ResizeObserver && overflowElement) {
-      this._resizeObserver = new ResizeObserver(this._async.debounce(this._checkTextOverflow, 500));
-      this._resizeObserver.observe(overflowElement);
+  const checkTextOverflow = useCallback(() => {
+    const overflowElement = getTargetElement();
+    const isTextOverflow = !!overflowElement && hasOverflow(overflowElement);
+    if (isTextOverflow !== textOverflow) {
+      setTextOverflow(isTextOverflow);
     }
-  }
+  }, [getTargetElement, textOverflow, setTextOverflow]);
 
-  public componentWillUnmount(): void {
-    this._resizeObserver?.disconnect();
-    this._async.dispose();
-  }
+  useEffect(() => {
+    checkTextOverflow();
+  }, [checkTextOverflow]);
 
-  private _checkTextOverflow = (): void => {
-    const overflowElement = this._getTargetElement();
-    const textOverflow = !!overflowElement && hasOverflow(overflowElement);
-    if (textOverflow !== this.state.textOverflow) {
-      this.setState({ textOverflow });
+  useEffect(() => {
+    const overflowElement = getTargetElement();
+    if (window.ResizeObserver && overflowElement) {
+      resizeObserver.current = new window.ResizeObserver(async.debounce(checkTextOverflow, 500));
+      resizeObserver.current.observe(overflowElement);
     }
-  };
 
-  private _getTargetElement = (): HTMLElement | undefined => {
-    if (!this._tooltipChild.current || !this._tooltipChild.current.parentElement) {
-      return undefined;
-    }
+    return () => {
+      resizeObserver.current?.disconnect();
```

Comment:
how are you mimicking unmount behavior in this effect

Reply:
The useEffect hook does not execute the entire effect on every lifecycle phase. Instead, it runs the effect on mount and update, and the cleanup function on unmount.
On Mount:
  - Call getTargetElement()
  - If ResizeObserver is available and overflowElement is not null:
    - Create new ResizeObserver with debounced checkTextOverflow
    - Observe overflowElement

On Update (when async, checkTextOverflow, or getTargetElement change):
  - Call getTargetElement()
  - If ResizeObserver is available and overflowElement is not null:
    - Create new ResizeObserver with debounced checkTextOverflow
    - Observe overflowElement

On Unmount:
  - Disconnect ResizeObserver
  - Dispose async

Reply:
Verified as follows: 
Setup/update: Setup block is called when the chart load. 
Unmount: Used a button to unmount the chart. when the button is clicked the cleanup block gets executed.

Reply:
I am not getting where is the logic to restricted to run on unmount. I get that the dispose logic is executed when resize observer disconnects. Lets discuss this more. Could you log a bug for this.
Meanwhile approving this PR to unblock the build.

Reply:
This article explains in detail https://react.dev/reference/react/useEffect#reference regarding how the setup and cleanup parts work within the useeffect.